From 610aa7bc418a9e177decf922620db909c3a0e34e Mon Sep 17 00:00:00 2001
Message-ID: <610aa7bc418a9e177decf922620db909c3a0e34e.1713654571.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1713654571.git.sharlatanus@gmail.com>
References: <cover.1713654571.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Wed, 3 Apr 2024 18:35:22 +0100
Subject: [PATCH 28/47] gnu: python-bayesicfitting: Refresh package style.

* gnu/packages/astronomy.scm (python-bayesicfitting): Adjust
indentation.
[build-system]: Swap to pyproject-build-system.
[arguments] <#:test-flags>: Add them.
<#:phases>: Use test flags in 'check phase.

Change-Id: Iffc8591199bbaf2a0da95b6b22476e34121dbe79
---
 gnu/packages/astronomy.scm | 42 ++++++++++++++++++++++----------------
 1 file changed, 24 insertions(+), 18 deletions(-)

diff --git a/gnu/packages/astronomy.scm b/gnu/packages/astronomy.scm
index bf71843e05..618d0e78bf 100644
--- a/gnu/packages/astronomy.scm
+++ b/gnu/packages/astronomy.scm
@@ -1934,25 +1934,31 @@ (define-public python-bayesicfitting
   (package
     (name "python-bayesicfitting")
     (version "3.2.0")
-    (source (origin
-              (method git-fetch)
-              (uri (git-reference
-                    (url "https://github.com/dokester/BayesicFitting")
-                    (commit (string-append "v" version))))
-              (file-name (git-file-name name version))
-              (sha256
-               (base32
-                "0q6v7w9b1xzam0qn5vsl5wrdp1fkfpsn411pzd8wyy9giznpajxi"))))
-    (build-system python-build-system)
+    (source
+     (origin
+       (method git-fetch)
+       (uri (git-reference
+             (url "https://github.com/dokester/BayesicFitting")
+             (commit (string-append "v" version))))
+       (file-name (git-file-name name version))
+       (sha256
+        (base32 "0q6v7w9b1xzam0qn5vsl5wrdp1fkfpsn411pzd8wyy9giznpajxi"))))
+    (build-system pyproject-build-system)
     (arguments
-     (list #:phases #~(modify-phases %standard-phases
-                        (replace 'check
-                          (lambda* (#:key tests? #:allow-other-keys)
-                            (when tests?
-                              (invoke "python" "-m" "unittest" "discover"
-                                      "test")))))))
-    (propagated-inputs (list python-astropy python-future python-matplotlib
-                             python-numpy python-scipy))
+     (list
+      #:test-flags #~(list "-m" "unittest" "discover" "test")
+      #:phases
+      #~(modify-phases %standard-phases
+          (replace 'check
+            (lambda* (#:key tests? test-flags #:allow-other-keys)
+              (when tests?
+                (apply invoke "python" test-flags)))))))
+    (propagated-inputs
+     (list python-astropy
+           python-future
+           python-matplotlib
+           python-numpy
+           python-scipy))
     (home-page "https://www.bayesicfitting.nl")
     (synopsis "Python Toolbox for Astronimical Bayesian fitting")
     (description
-- 
2.41.0

